public/app/core/components/PageBanner/PageBanner.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState, useCallback, useMemo } from 'react';
+import { Icon, IconButton, IconName, LinkButton, useStyles2, useTheme2 } from '@grafana/ui';
+import { css } from '@emotion/css';
+import { GrafanaTheme2, PageBannerDisplayEvent, PageBannerEventPayload, PageBannerSeverity } from '@grafana/data';
+import { appEvents } from 'app/core/core';
+
+export function PageBanner(): React.ReactElement | null {
+  const [banner, setBanner] = useState<PageBannerEventPayload | undefined>();
+  const severityStyling = useStylingBySeverity(banner?.severity);
+  const styles = useStyles2((theme) => getStyles(theme, severityStyling));
+  const onClose = useCallback(() => setBanner(undefined), []);
+
+  useEffect(() => {
+    const subscription = appEvents.subscribe(PageBannerDisplayEvent, (event) => {
+      setBanner(event.payload);
+    });
+    return subscription.unsubscribe;
+  }, []);
```

Comment:
```suggestion
  }, [setBanner]);
```

Reply:
@

> think this looks good
> 
> * we need to add appEvents to @grafana/runtime
> * implement closable
> 
> The top level css changs makes this a bit risky to backport directly into 8.2, but maybe having this in main soon is good enough for cloud to start implementation

I added a ticket for exposing appEvents (https://github.com/grafana/grafana/issues/39276) but I guess in the first MVP of this feature it might be sufficient with the current way of consuming the appEvents.